isis/src/base/objs/CSMCamera/CSMCamera.cpp

```diff
@@ -566,8 +534,15 @@ namespace Isis {
 
 
   double CSMCamera::IncidenceAngle() const {
-    csm::EcefVector sunEcefVec = m_model->getIlluminationDirection(isisToCsmGround(GetSurfacePoint()));
-    std::vector<double> sunVec = {sunEcefVec.x, sunEcefVec.y, sunEcefVec.z};
+    csm::EcefCoord groundPt = isisToCsmGround(GetSurfacePoint());
+    csm::EcefVector sunEcefVec = m_model->getIlluminationDirection(groundPt);
+    // ISIS wants the position of the sun, not just the vector from the ground
+    // point to the sun. So, we approximate this by adding in the ground point.
+    // ISIS wants this in Km so convert
+    std::vector<double> sunVec = {
+        (sunEcefVec.x + groundPt.x) / 1000.0,
+        (sunEcefVec.y + groundPt.y) / 1000.0,
+        (sunEcefVec.z + groundPt.z) / 1000.0};
```

Comment:
Even with this fix, the USGSCSM phase and incidence angles are not matching, we may need to do some more digging on the USGSCSM side.

Reply:
Turns out we had the vector backwards, from the CSM docs:

    ﻿This method returns a vector defining the direction of illumination at the given ground position that points from the illumination source to the given ground point.